builder/static_code.go

```diff
@@ -888,41 +951,208 @@ func listJoin(list []string, sep string, lastSep string) string {
 	}
 }
 
-func (p *parser) parseRule(rule *rule) (any, bool) {
+// ==template== {{ if .LeftRecursion }}
+
+func (p *parser) checkPrevChoice(rule *rule) (bool, bool) {
+	var currentEStack []any
+	var prevEStack []any
+	for i := 1; i <= len(p.rstack); i++ {
+		indexCurrent := len(p.rstack) - i
+		indexPrev := len(p.rstack) - i*2
```

Comment:
Because there is a recursion search. Between the same elements should be the same distance.
```
 A -> A -> A
[0]  [1]  [2]
 A -> B -> A -> B -> A
[0]       [2]       [4]                
 A -> B -> C -> A -> B -> C -> A
[0]            [3]            [6]
```

Reply:
Understood, thanks for the example and the explanation.